gptp/layers.py

```diff
@@ -25,42 +25,49 @@ class PTPv2(Packet):
     ]
 
     fields_desc = [
-        BitField("transportSpecific", 1, 4),
+        BitField("majorSdoId", 1, 4),
         BitEnumField("messageType", 0, 4, MSG_TYPES),
-        XBitField("reserved0", 0, 4),
+        XBitField("minorVersionPTP", 0, 4),
         BitField("versionPTP", 0x2, 4),
         ShortField("messageLength", 34),
         ByteField("domainNumber", 0),
-        XByteField("reserved1", 0),
+        XByteField("minorSdoId", 0),
         FlagsField("flags", 0, 16, FLAGS),
         LongField("correctionField", 0),
-        XIntField("reserved2", 0),
+        XIntField("messageTypeSpecific", 0),
         PortIdentityField("sourcePortIdentity", 0),
         ShortField("sequenceId", 0),
-        XByteField("control", 0),
+        XByteField("controlField", 0),
         SignedByteField("logMessageInterval", -3),
 
-        # Sync
-        ConditionalField(BitField("reserved3", 0, 80), lambda pkt: pkt.is_sync),
+        # Sync (twoStep flag not set) or PdelayReq
+        ConditionalField(
+            BitField("reserved", 0, 80),
+            lambda pkt: (pkt.is_sync and not pkt.twostepflag_set) or pkt.is_pdelay_req),
```

Comment:
IEEE Std 802.1AS-2020 (chapter 11.4.3.1) explicitely differentiates the sync message format, in dependence of the two step flag.

In case two step flag is set the 'reserved' field is present, in case it is not set, 'originTimestamp' (same size as 'reserved') and 'Follow_Up information TLV' are used. 

In 802.1AS-2011 this differentiation was not present.